src/char.cr

```diff
@@ -92,14 +92,24 @@ struct Char
     self - other
   end
 
-  # Returns `true` if this char is an ASCII digit ('0' to '9').
+  # Returns `true` if this char is an ASCII digit in specified base.
+  #
+  # Base can be from 0 to 36 with digits from '0' to '9' and 'a' to 'z' or 'A' to 'Z'.
   #
   # ```
-  # '4'.digit? # => true
-  # 'z'.digit? # => false
+  # '4'.digit?     # => true
+  # 'z'.digit?     # => false
+  # 'z'.digit?(36) # => true
   # ```
-  def digit?
-    '0' <= self <= '9'
+  def digit?(base : Int = 10)
+    raise ArgumentError.new "invalid base #{base}, expected 2 to 36" unless 2 <= base <= 36
+    ord = ord()
+    if ord < 256
+      digit = String::CHAR_TO_DIGIT.to_unsafe[ord]
+      digit != -1 && digit < base
+    else
+      false
+    end
```

Comment:
Actually since the implementation became nearly identical, how about reusing `to_i`? `to_i(base) { false } || true`